src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -5982,6 +5982,99 @@ MergeTreeData::MutableDataPartPtr MergeTreeData::cloneAndLoadDataPartOnSameDisk(
     return dst_data_part;
 }
 
+MergeTreeData::MutableDataPartPtr MergeTreeData::cloneAndLoadPartOnSameDiskWithDifferentPartitionKey(
```

Comment:
This method is a copy of `cloneAndLoadPartOnSameDisk`, but with some key modifications. See diff: https://www.diffchecker.com/CnJqCC18.

It hard links all files, just like the original method, but then deletes/unlink `checksums.txt`, `partition.dat` and `minmax_<partition_exp_columns>`. It then writes the updated files to destination table partition directory.